transcriptic/commands.py

```diff
@@ -865,6 +867,90 @@ def launch(api, protocol, project, save_input, local, accept_quote, params, pm=N
                 api, manifest, protocol_obj, inputs
             )
 
+def generate_preview_parameters(api, protocol, project, local, filename, merge, pkg=None):
+    """Generate preview parameters for a specified protocol from selected samples"""
+    # Load protocol from local file if not remote and load from listed protocols otherwise
+    if local:
+        manifest, protocol_obj = load_manifest_and_protocol(protocol)
+    else:
+        print_stderr("Searching for {}...".format(protocol))
+        protocol_list = api.get_protocols()
+
+        matched_protocols = [p for p in protocol_list if (
+            p['name'] == protocol and (pkg is None or p['package_id'] == pkg)
+        )]
+
+        if len(matched_protocols) == 0:
+            print_stderr(
+                "Protocol {} in {} was not found."
+                "".format(
+                    protocol,
+                    "package {}".format(pkg) if pkg else "unspecified package"
+                )
+            )
+            return
+        elif len(matched_protocols) > 1:
+            print_stderr("More than one match found. Using the first match.")
+        else:
+            print_stderr("Protocol found.")
+        protocol_obj = matched_protocols[0]
+    # Project is required for quick launch
+    if not project:
+        click.echo("Project field is required if parameters file is not specified.")
+        return
+    project = get_project_id(api, project)
+    if not project:
+        return
+    # Creates web browser and generates inputs for quick_launch
+    quick_launch = _get_quick_launch(api, protocol_obj, project)
+    pp = PreviewParameters(api, quick_launch["raw_inputs"])
+    # Determine where to place the generated preview parameters
+    if not merge:
+        if not filename:
+            filename = 'preview_parameters.json'
+        try:
+            with click.open_file(filename, 'w') as f:
+                f.write(json.dumps(pp.preview, indent=2))
+                f.close()
```

Comment:
should there be an else statement here? if not, seems like filename should be required?